core/domain/classifier_services.py

```diff
@@ -127,6 +127,69 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
+
+
+def add_to_training_queue(exploration):
+    """Checks all states of the exploration for re-training conditions and
+    creates ClassifierTrainingJobModel instances in the datastore
+    accordingly.
+
+    Args:
+        exploration: Exploration. Domain object for an exploration.
+    """
+    states = exploration.states
+    for state_name in states:
+        state = states[state_name]
+        if not state.can_undergo_classification():
+            continue
+        exp_id = exploration.id
+        exp_version = exploration.version
+        training_data = state.get_training_data()
+        interaction_id = state.interaction.id
+        algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[
+            interaction_id]['algorithm_id']
+        classifier = get_classifier_from_exploration_attributes(
+            exp_id, exp_version, state_name)
+        if classifier:
```

Comment:
AFAICT, the version for the mapping is incremented but the version for the job is not?

That said, doing all this ad hoc (or spreading implicit info of the version throughout the code) is going to make the code more unmaintainable. I would suggest just passing the correct version in from the get-go and using it throughout.